src/backend/drm/mod.rs

```diff
@@ -57,335 +50,118 @@
-        mode: Mode,
-        connectors: &[connector::Handle],
-    ) -> Result<Self::Surface, <Self::Surface as Surface>::Error>;
-
-    /// Processes any open events of the underlying file descriptor.
-    ///
-    /// You should not call this function manually, but rather use
-    /// [`device_bind`] to register the device
-    /// to an [`EventLoop`](calloop::EventLoop)
-    /// to synchronize your rendering to the vblank events of the open crtc's
-    fn process_events(&mut self);
-
-    /// Attempts to acquire a copy of the [`Device`]'s
-    /// [`ResourceHandle`](drm::control::ResourceHandle)
-    fn resource_handles(&self) -> Result<ResourceHandles, <Self::Surface as Surface>::Error>;
-
-    /// Retrieve the information for a connector
-    fn get_connector_info(&self, conn: connector::Handle) -> Result<connector::Info, DrmError>;
-
-    /// Retrieve the information for a crtc
-    fn get_crtc_info(&self, crtc: crtc::Handle) -> Result<crtc::Info, DrmError>;
-
-    /// Retrieve the information for an encoder
-    fn get_encoder_info(&self, enc: encoder::Handle) -> Result<encoder::Info, DrmError>;
-
-    /// Retrieve the information for a framebuffer
-    fn get_framebuffer_info(&self, fb: framebuffer::Handle) -> Result<framebuffer::Info, DrmError>;
-
-    /// Retrieve the information for a plane
-    fn get_plane_info(&self, plane: plane::Handle) -> Result<plane::Info, DrmError>;
+//! A commit/page_flip may be triggered to apply the pending state.
+//!
+//! ## Rendering
+//!
+//! The drm infrastructure makes no assumptions about the used renderer and does not interface with them directly.
+//! It just provides a way to create framebuffers from various buffer types (mainly `DumbBuffer`s and hardware-backed gbm `BufferObject`s).
+//!
+//! Buffer management and details about the various types can be found in the [`allocator`-Module](backend::allocator) and
+//! renderering abstractions, which can target these buffers can be found in the [`renderer`-Module](backend::renderer).
+
+pub(crate) mod device;
+pub(self) mod error;
+#[cfg(feature = "backend_gbm")]
+mod render;
+#[cfg(feature = "backend_session")]
+pub(self) mod session;
+pub(self) mod surface;
+
+pub use device::{device_bind, DevPath, DeviceHandler, DrmDevice, DrmSource};
+pub use error::Error as DrmError;
+#[cfg(feature = "backend_gbm")]
+pub use render::{DrmRenderSurface, Error as DrmRenderError};
+#[cfg(feature = "backend_session")]
+pub use session::{DrmDeviceObserver, DrmSurfaceObserver};
+pub use surface::DrmSurface;
+
+use drm::control::{plane, crtc, Device as ControlDevice, PlaneType};
+
+/// A set of planes as supported by a crtc
+pub struct Planes {
+    /// The primary plane of the crtc (automatically selected for [DrmDevice::create_surface])
+    pub primary: plane::Handle,
+    /// The cursor plane of the crtc, if available
+    pub cursor: Option<plane::Handle>,
+    /// Overlay planes supported by the crtc, if available
+    pub overlay: Option<Vec<plane::Handle>>,
```

Comment:
Is the `Option<_>` wrapping necessary here? I't think that an empty `Vec` could play the role of `None`. :thinking: